### PR TITLE
Cart/addcart

### PR DESCRIPTION
**Pull request by ninhbuise**
Customer can add product to cart #93

### CODE REVIEW

File: cart/src/main/java/com/yas/cart/controller/CartController.java [hunk 1]
```diff
@@ -34,4 +45,21 @@ public ResponseEntity<List<CartGetDetailVM>> listCartDetailByCustomerId(@PathVar
             return ResponseEntity.ok(cartService.getCartDetailByCustomerId(customerId));
         return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(null);
     }
+
+    @PostMapping(path = "/storefront/carts")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Created", content = @Content(schema = @Schema(implementation = CartGetDetailVM.class))),
+            @ApiResponse(responseCode = "400", description = "Bad request", content = @Content(schema = @Schema(implementation = ErrorVm.class))) })
+    public ResponseEntity<CartGetDetailVM> createCart(@Valid @RequestBody CartPostVM cartPostVM,
```
Comment (thiennn) [inline cart/src/main/java/com/yas/cart/controller/CartController.java:53]: Customer need to have access token to call this api

Comment (thiennn) [reply cart/src/main/java/com/yas/cart/controller/CartController.java:53]: Only customer can create cart for himself, don't need to check admin role

File: cart/src/main/java/com/yas/cart/service/CartService.java [hunk 2]
```diff
@@ -29,4 +39,39 @@ public List<CartGetDetailVM> getCartDetailByCustomerId(String customerId) {
                 .stream().map(CartGetDetailVM::fromModel)
                 .toList();
     }
+
+    public CartGetDetailVM createCart(CartPostVM cartPostVM) {
```
Comment (thiennn) [inline cart/src/main/java/com/yas/cart/service/CartService.java:43]: Vm not VM

File: cart/src/main/java/com/yas/cart/viewmodel/CartDetailPostVM.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.yas.cart.viewmodel;
+
+import com.yas.cart.model.CartDetail;
+
+public record CartDetailPostVM(Long productId, int quantity) {
```
Comment (thiennn) [inline cart/src/main/java/com/yas/cart/viewmodel/CartDetailPostVM.java:5]: CartItemVm seem better name

File: cart/src/main/java/com/yas/cart/viewmodel/CartPostVM.java [hunk 4]
```diff
@@ -5,7 +5,6 @@
 import javax.validation.constraints.NotEmpty;
 
 public record CartPostVM(
-        @NotEmpty Long id,
         @NotEmpty String customerId,
-        List<CartDetailListVM> cartDetailListVMs) {
+        List<CartDetailPostVM> CartDetailPostVMs) {
```
Comment (thiennn) [inline cart/src/main/java/com/yas/cart/viewmodel/CartPostVM.java:9]: user camel case for parameter name

File: cart/src/main/java/com/yas/cart/service/CartService.java [hunk 5]
```diff
@@ -29,4 +39,39 @@ public List<CartGetDetailVM> getCartDetailByCustomerId(String customerId) {
...
+        cart.setCustomerId(cartPostVM.customerId());
+        if (CollectionUtils.isNotEmpty(cartPostVM.CartDetailPostVMs())) {
+            for (CartDetailPostVM cartDetailPostVM : cartPostVM.CartDetailPostVMs()) {
+                try {
+                    productService.getProduct(cartDetailPostVM.productId());
+                } catch (Exception e) {
+                    throw new BadRequestException(String.format("Not found product %d", cartDetailPostVM.productId()));
+                }
+                
+                if(cartDetailPostVM.quantity() <= 0)
```
Comment (thiennn) [inline cart/src/main/java/com/yas/cart/service/CartService.java:56]: move this check before productService.getProduct for more efficiency

**Review by thiennn — CHANGES_REQUESTED** [review]

File: cart/src/main/java/com/yas/cart/controller/CartController.java (hunk 1 above)
Comment (ninhbuise) [reply cart/src/main/java/com/yas/cart/controller/CartController.java]: Any request add to cart need login 1st (access token)

**Review by thiennn — APPROVED** [review]

**Pull request merged**